[TiewJiaLiang] iP

TiewJiaLiang opened the pull request:
### WizT

> "The function of good software is to make the complex appear to be simple" - Grady Booch ([source](https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.azquotes.com%2Fquote%2F602639&psig=AOvVaw2qqBhfBCAJAQ-4tzKhumlb&ust=1738478719508000&source=images&cd=vfe&opi=89978449&ved=0CBcQjhxqFwoTCLiF_qXwoYsDFQAAAAAdAAAAABAE))

WizT: The all in one platform to help you track your tasks. It's:
- Text-Based
- Simple to learn
- ~QUICK~ _VERY_ QUICK to use

All you need to do is

1. Download it 
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you! 🥳 

And guess what? It is **FREEEEEE!**

Features:

- [x] Managing Tasks
- [ ] Managing Deadlines (Stay Tuned!)
- [ ] Reminders (Stay Tuned!)

If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TiewJiaLiang committed:
Rename to WizT

TiewJiaLiang committed:
Added Echo

TiewJiaLiang committed:
Added storing function

TiewJiaLiang committed:
Added Task Class and marking function

TiewJiaLiang committed:
Added Subclasses

TiewJiaLiang committed:
Added automated testing files

TiewJiaLiang committed:
Added Error Handling

TiewJiaLiang committed:
Added delete function

TiewJiaLiang committed:
Fixed bye function

TiewJiaLiang committed:
Added name and Student Num

TiewJiaLiang committed:
Add save function

TiewJiaLiang committed:
Add wizt.txt to master branch

TiewJiaLiang committed:
Merge branch 'branch-Level-7'

TiewJiaLiang committed:
Change date format

TiewJiaLiang committed:
Add OOP classes

TiewJiaLiang committed:
Separated into packages

TiewJiaLiang committed:
Merge branch 'add-gradle-support'

TiewJiaLiang committed:
Set up Gradle

TiewJiaLiang committed:
Creat JUnit tests

TiewJiaLiang committed:
Package app as jar file

TiewJiaLiang committed:
Add JavaDoc comments

TiewJiaLiang committed:
Tweak code to comply with standard

TiewJiaLiang committed:
Add find function

TiewJiaLiang committed:
Merge branch 'branch-A-JavaDoc'

TiewJiaLiang committed:
Merge branch 'branch-A-CodingStandard'

TiewJiaLiang committed:
Merge branch 'branch-Level-9'

TiewJiaLiang committed:
Edit JavaDoc

TiewJiaLiang committed:
Fix code style

TiewJiaLiang committed:
Merge branch 'branch-A-CheckStyle'

TiewJiaLiang committed:
Add GUI

TiewJiaLiang committed:
Merge branch 'branch-Level-10'

kumar2215 commented on gradle/wrapper/gradle-wrapper.jar:
Accidentally committed .jar file?

kumar2215 commented on src/main/java/wizt/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+
+    }
+
+    /*public void execute(TaskList tasks, Ui ui, Storage storage) throws WizTException {
```
Did you perhaps forget to remove this before committing?

kumar2215 commented on src/main/java/wizt/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+
+    }*/
+
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws WizTException {
```
If there is no implementation meant here, perhaps it should be labelled abstract?

kumar2215 commented on src/main/java/wizt/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ *  Represents any commands that user inputs into program
+ */
+public class Command {
```
Perhaps you meant this class to be abstract if you were following the guidelines?

![image](https://github.com/user-attachments/assets/6323bdbe-ebff-480f-bc69-3eaba00983e7)

kumar2215 commented on src/main/java/wizt/ui/WizT.java:
```diff
@@ -0,0 +1,117 @@
+     * @param filename  file name of the tasklist
+     * @param al Arraylist of all the tasks
+     */
+    public static void writeToFile(String filename, ArrayList<Task> al) {
```
Perhaps it makes more sense to put this functionality in the storage class?

kumar2215 commented on src/main/java/wizt/ui/WizT.java:
```diff
@@ -0,0 +1,117 @@
+     * The main program execution
+     * @throws WizTException
+     */
+    public void run() throws WizTException {
```
Why does the run method throw `WizTException` if none of the code outside the try block throws it?

kumar2215 commented on src/main/java/wizt/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+ *  Represents any commands that user inputs into program
+ */
+public class Command {
+    public Command() {
```
I don't think an explicit empty constructor is necessary here since Java automatically provides one. Perhaps consider one that initializes the input to the command? This way the child command classes don't have to keep an empty constructor too.

kumar2215 commented on src/main/java/wizt/ui/WizT.java:
```diff
@@ -0,0 +1,117 @@
+        }
+    }
+
+    public static void main(String[] args) throws WizTException {
```
I think main methods in general should not throw an exception. In this case, as stated for the run() method, there isn't a need for it here too.

kumar2215 commented on src/main/java/wizt/ui/WizTException.java:
Shouldn't this be in its own exceptions package and not the ui one?

kumar2215 commented on src/main/java/wizt/parser/Parser.java:
```diff
@@ -0,0 +1,67 @@
+     * @return the corresponding command type
+     */
+    public static Command parse(String fullCommand) {
+        if (fullCommand.equals("list")) {
```
Perhaps you could restructure this? `else if` is a thing in Java.

kumar2215 commented on src/main/java/wizt/parser/Parser.java:
```diff
@@ -0,0 +1,67 @@
+        if (fullCommand.equals("list")) {
+            return new ListCommand();
+        } else {
+            if (fullCommand.contains("unmark")) {
```
Is containing the right keyword a sufficient condition for it to be in the right format of a command? Or do the Command classes deal with cases where it contains the keyword but not in the right format (Eg. "pls unmark 3")?

kumar2215 commented on src/main/java/wizt/command/UpdateCommand.java:
```diff
@@ -0,0 +1,60 @@
+                    .append("-------------------------------------");
+        } else {
+
+            if (input1.contains("mark")) {
```
Maybe consider separating this into 2 command classes: MarkCommand and UnmarkCommand?

kumar2215 commented on src/main/java/wizt/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+            }
+            Task t = new Todo(substr);
+            al.add(t);
+            response.append("\n -------------------------------------")
```
Some of these lines being appended here are being repeated in each of the different task types. Perhaps consider putting the string representations of each task type in their individual subclasses and then have a generic way of creating the response using them.

kumar2215 commented on src/main/java/wizt/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+     */
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws WizTException {
+        ArrayList<Task> al = tasks.getTasksList();
```
Variable name `al` doesn't seem to be meaningful here. Perhaps it could be renamed to taskList or not be created at all?

kumar2215 commented on src/main/java/wizt/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+
+            //format dd/MM/YYYY HHmm
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+            LocalDateTime dt = LocalDateTime.parse(as[1], formatter);
```
A `DateTimeParseException` can be thrown here but doesn't seem to be handled.

kumar2215 commented on src/main/java/wizt/command/AddCommand.java:
```diff
@@ -0,0 +1,97 @@
+                if (substr.isEmpty()) {
+                    throw new WizTException("Please enter a time period!");
+                }
+                String[] as = substr.split(" /from");
```
Perhaps you can consider the case where the string is not in the right format?

kumar2215 commented on src/main/java/wizt/ui/WizT.java:
```diff
@@ -0,0 +1,117 @@
+    }
+
+
+    public String getResponse(String fullCommand) throws WizTException {
```
The only method call for this method is in this file and is being commented out. Perhaps you can consider removing it if it's not being used?

kumar2215 commented on src/main/java/wizt/ui/WizT.java:
```diff
@@ -0,0 +1,117 @@
+               // getResponse(fullCommand);
+
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
```
Each of the command classes have String as the return type of their execute method but that is not being used here. Perhaps consider making them have a return type of void then?

kumar2215 commented on src/main/java/wizt/command/UpdateCommand.java:
```diff
@@ -0,0 +1,60 @@
+        if (input1.contains("unmark")) {
+            String[]split = input1.split(" ");
+
+            int no = Integer.parseInt(split[1]);
```
`NumberFormatException` can be thrown here if the format of the string is not what is expected to be. Perhaps consider handling it.

kumar2215 commented on src/main/java/wizt/command/UpdateCommand.java:
```diff
@@ -0,0 +1,60 @@
+            if (input1.contains("mark")) {
+                String[]split = input1.split(" ");
+                int no = Integer.parseInt(split[1]);
+                al.get(no - 1).markAsDone();
```
Shouldn't the TaskList class be responsible for marking, unmarking, deleting and adding of tasks? I don't think it's a good idea to directly access the task here in any of the command classes.

kumar2215 reviewed (commented):
I believe the quality of code can be improved here. There are a lot of places where exceptions can be thrown but are not handled.

xa0412 commented on src/main/java/wizt/task/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+ */
+public class Deadline extends Task {
+
+    protected LocalDateTime by;
```
Would it be better to make LocalDateTime private and accessed via a getter to enforce encapsulation?

xa0412 commented on src/main/java/wizt/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+        File f = new File(filename);
+        try {
+            if (f.createNewFile()) {
+                System.out.println("File dont exist!, created a new file wizt.txt!");
```
(File does not exist!) rather then (File dont exist!)?

xa0412 commented on src/main/java/wizt/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+
+
+        File filename = new File(this.filename);
+        Scanner s = new Scanner(filename);
+        ArrayList<Task> al = new ArrayList<>();
```
Use meaningful variable name?
s -> scanner
al -> tasks

xa0412 commented on src/main/java/wizt/ui/Main.java:
```diff
@@ -0,0 +1,36 @@
+ */
+public class Main extends Application {
+
+    private WizT wizt = new WizT("wizt.txt");
```
Since "wizt.txt" is the default file name, store it in a constant?
private static final String FILENAME = "wizt.txt";

xa0412 commented on src/main/java/wizt/storage/Storage.java:
```diff
@@ -0,0 +1,94 @@
+            if (line.contains("from")) {
+                line = line.substring(7);
+
+                String[] as = line.split("\\(from:");
+                String[] as2 = as[1].split("to:");
```
Same: Use meaningful variable name?

xa0412 reviewed (commented):
Looks good, just some advice from me.

TiewJiaLiang committed:
Fix update issues to wiz.txt

The code was not updating the wiz.txt for update commands

This needs to be changed since if changes are not saved into the text file, the subsequent iteration of the program will not output the correct tasklist

TiewJiaLiang committed:
Add assert feature

TiewJiaLiang committed:
Improve code quality

TiewJiaLiang committed:
Add streams

TiewJiaLiang committed:
Merge pull request #1 from TiewJiaLiang/branch-A-Assertions

Add assert feature

TiewJiaLiang committed:
Merge branch 'branch-A-CodeQuality'

# Conflicts:
#	src/main/java/wizt/command/AddCommand.java
#	src/main/java/wizt/command/DeleteCommand.java
#	src/main/java/wizt/command/UpdateCommand.java
#	src/main/java/wizt/ui/MainWindow.java

TiewJiaLiang committed:
Merge branch 'branch-A-Streams'

# Conflicts:
#	src/main/java/wizt/command/FindCommand.java

TiewJiaLiang committed:
Add extension

TiewJiaLiang committed:
Merge branch 'branch-BCD-Extension'

TiewJiaLiang committed:
Setup CI

TiewJiaLiang committed:
Rename variable

TiewJiaLiang committed:
Improve GUI

TiewJiaLiang committed:
Give Personality to App

TiewJiaLiang committed:
Add more error handling

TiewJiaLiang committed:
Add more testing

TiewJiaLiang committed:
Update ReadMe

TiewJiaLiang committed:
Update ReadME

TiewJiaLiang committed:
Fix bat file

TiewJiaLiang committed:
Fix bat file

TiewJiaLiang committed:
Fix Checkstyle

TiewJiaLiang committed:
Fix bat file

TiewJiaLiang committed:
Fix bat and checkstyle

TiewJiaLiang committed:
Fix bat

TiewJiaLiang committed:
Fix bat

TiewJiaLiang committed:
Fix bat

TiewJiaLiang committed:
Fix sh

TiewJiaLiang committed:
Update ReadMe

TiewJiaLiang committed:
Rename Image

TiewJiaLiang committed:
Update error handling

TiewJiaLiang committed:
Fix error handling

TiewJiaLiang committed:
Modify image to conform to standards

TiewJiaLiang committed:
Fix quality

TiewJiaLiang committed:
Fix issue with event and todo

TiewJiaLiang committed:
Remove debug statement

TiewJiaLiang committed:
Update Readme
